src/message.rs

```diff
@@ -1169,9 +1169,17 @@ pub async fn get_msg_info(context: &Context, msg_id: MsgId) -> Result<String> {
     }
     if let Some(ref server_folder) = msg.server_folder {
         if !server_folder.is_empty() {
-            ret += &format!("\nLast seen as: {}/{}", server_folder, msg.server_uid);
+            ret += &format!("\nLast seen as: {}/{}\n", server_folder, msg.server_uid);
         }
     }
+    let hop_info: Option<String> = context
+        .sql
+        .query_get_value("SELECT hop_info FROM msgs WHERE id=?;", paramsv![msg_id])
+        .await?;
+
+    if hop_info.is_some() {
+        ret.push_str(&hop_info.unwrap_or_else(|| "No Hop info".to_owned()));
+    }
```

Comment:
- `"No Hop info"` is never shown as you check `if hop_info.is_some()`.
- instead of `push_str()` you could use `+=` here, just as above
- I'd add a "\n" before the first line. Currently there are two new lines inbetween the individual hop infos, but just one before the hop info:
```
Message-ID: abcd@gmx.de
Last seen as: INBOX/13744
1. Hop:                                 <-- I think there should be an extra newline before this line
Date: Mon, 20 Dec 2021 18:34:12 +0100
From: [192.168.0.230]
By: mail.gmx.net

2. Hop:
Date: Mon, 20 Dec 2021 18:34:13 +0100
From: mout.gmx.net
By: hq5.merlinux.eu

3. Hop:
Date: Mon, 20 Dec 2021 18:34:13 +0100
From: hq5.merlinux.eu
By: hq5.merlinux.eu
```